src/Host/Monitor/Impl/CredentialManager.cs

```diff
@@ -172,6 +175,10 @@ public class CredentialManager {
                     psi.Domain = domainBldr.ToString();
                     psi.Password = pass;
                 }
+
+                if (cred != IntPtr.Zero) {
+                    CredFree(cred);
```

Comment:
This should be inside `finally`, since some code paths above might not get here (e.g. if `CredUIParseUserName` fails, we throw).